core/lib/types/src/snapshots.rs

```diff
@@ -3,28 +3,47 @@ use zksync_basic_types::{L1BatchNumber, MiniblockNumber};
 
 use crate::{commitment::L1BatchWithMetadata, Bytes, StorageKey, StorageValue};
 
+/// Information about all snapshots persisted by the node.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct AllSnapshots {
+    /// L1 batch numbers for complete snapshots. Ordered by descending number (i.e., 0th element
+    /// corresponds to the newest snapshot).
     pub snapshots_l1_batch_numbers: Vec<L1BatchNumber>,
+    /// Same as `snapshots_l1_batch_numbers`, but for incomplete snapshots (ones in which some
+    /// storage log chunks may be missing).
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub incomplete_snapshots_l1_batch_numbers: Vec<L1BatchNumber>,
 }
 
-// used in dal to fetch certain snapshot data
-#[derive(Debug, Clone, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
+/// Storage snapshot metadata. Used in DAL to fetch certain snapshot data.
+#[derive(Debug, Clone)]
 pub struct SnapshotMetadata {
+    /// L1 batch for the snapshot. The data in the snapshot captures node storage at the end of this batch.
     pub l1_batch_number: L1BatchNumber,
+    /// Path to the factory dependencies blob.
     pub factory_deps_filepath: String,
-    pub storage_logs_filepaths: Vec<String>,
+    /// Paths to the storage log blobs. Ordered by the chunk ID. If a certain chunk is not produced yet,
+    /// the corresponding path is `None`.
+    pub storage_logs_filepaths: Vec<Option<String>>,
+}
+
+impl SnapshotMetadata {
+    /// Checks whether a snapshot is complete (contains all information to restore from).
+    pub fn is_complete(&self) -> bool {
+        self.storage_logs_filepaths.iter().all(Option::is_some)
+    }
 }
 
-//contains all data not contained in factory_deps/storage_logs files to perform restore process
+/// Snapshot data returned by using JSON-RPC API.
+/// Contains all data not contained in factory deps / storage logs files to perform restore process.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct SnapshotHeader {
     pub l1_batch_number: L1BatchNumber,
     pub miniblock_number: MiniblockNumber,
-    //ordered by chunk ids
+    pub is_complete: bool,
```

Comment:
As I'm thinking of it longer, can we make the API transparent to the resilience in snapshots creator? i.e. we remove this field and behave as if incomplete snapshots don't exist in RPC API?

Reply:
Yes, I think this makes sense at least initially 👍